cmd/qat_plugin/qat_plugin_test.go

```diff
@@ -362,3 +365,40 @@ func TestScanPrivate(t *testing.T) {
 		}
 	}
 }
+
+func TestPostAllocate(t *testing.T) {
+	response := new(pluginapi.AllocateResponse)
+	cresp := new(pluginapi.ContainerAllocateResponse)
+	response.ContainerResponses = append(response.ContainerResponses, cresp)
+	testMap := map[string]string{
+		"QAT29": "03:04.1",
+		"QAT13": "03:04.2",
+		"QAT6": "03:04.3",
+		"QAT21": "03:04.4",
+		}
+	response.ContainerResponses[0].Envs = testMap
+	resultKey := []string{
+		"QAT0",
+		"QAT1",
+		"QAT2",
+		"QAT3",
+	}
+	dp := &devicePlugin{}
+	dp.PostAllocate(response)
+	if len(response.ContainerResponses[0].Envs) != 4 {
+		t.Fatal("Set wrong number of Environment Variables")
+	}
+	counter := 0
+	for k := range response.ContainerResponses[0].Envs {
+			if k != resultKey[counter] {
```

Comment:
Since maps are implemented as hashes in Go lang their iteration order is not specified. This comparison may fail even if there's no bug in the code under test. Would be better to iterate over `resultKey` and check if the value from the response's `Envs` can be found in the expected set of values.
```
expectedValues := map[string]struct{}{
    "03:04.1": struct{}{},
   ...
}
for _, key := range resultKey {
    if value := response.ContainerResponses[0].Envs[key] {
        if _, ok := expectedValues[value]; ok {
            delete(expectedValues, value)
        } else {
            t.Errorf("Unexpected value %s", value)
        }
    }
}
```